src/Features/Core/Portable/GenerateType/AbstractGenerateTypeService.GenerateNamedType.cs

```diff
@@ -42,7 +42,7 @@ private async Task<INamedTypeSymbol> GenerateNamedTypeAsync(GenerateTypeOptionsR
         {
             if (options.TypeKind == TypeKind.Delegate)
             {
-                return CodeGenerationSymbolFactory.CreateDelegateTypeSymbol(
+                return (INamedTypeSymbol)CodeGenerationSymbolFactory.CreateDelegateTypeSymbol(
```

Comment:
why did we lose the type safety here?

Reply:
`CodeGenerationSymbolFactory.CreateDelegateTypeSymbol` needs to either return `CodeGenerationDelegateTypeSymbol` (which does not implement `INamedTypeSymbol`), or it needs to return `INamedTypeSymbol` (which does not extend `CodeGenerationDelegateTypeSymbol`). Either way, some code is going to end up with explicit casts.

Reply:
> CodeGenerationSymbolFactory.CreateDelegateTypeSymbol needs to either return CodeGenerationDelegateTypeSymbol (which does not implement INamedTypeSymbol)

If it doesn't implement INamedTypeSymbol, how can it be casted to taht type?

Reply:
This is an abstract type. All concrete types derived from it do implement `INamedTypeSymbol`.